### PR TITLE
doc diskio: RIOTized style and added doxygen

### PR DESCRIPTION
**Pull request by OlegHahm**
Just another minor step towards #2137.

### CODE REVIEW

File: drivers/include/diskio.h [hunk 1]
```diff
@@ -29,105 +30,141 @@
...
-/* Generic ioctl command (defined for FatFs) */
-#define CTRL_SYNC           0   /* Flush disk cache (for write functions) */
-#define GET_SECTOR_COUNT    1   /* Get media size (for only f_mkfs()) */
-#define GET_SECTOR_SIZE     2   /* Get sector size (for multiple sector size (_MAX_SS >= 1024)) */
-#define GET_BLOCK_SIZE      3   /* Get erase block size (for only f_mkfs()) */
-#define CTRL_ERASE_SECTOR   4   /* Force erased a block of sectors (for only _USE_ERASE) */
+/**
+ * @name Command code for disk_ioctrl fucntion
+ * @{
+ */
```
Comment (PeterKietzmann) [inline drivers/include/diskio.h]: Do I see it correctly that this is closed in line 112? If so, is this wanted?

**Comment by PeterKietzmann** [comment]
Mr. M found some issues that seem related to this PR. Please have a look Mr. Oleg

**Comment by OlegHahm** [comment]
Fixed CI findings.

@PeterKietzmann, most of this PR is search and replace. There should be no semantical changes.

**Comment by PeterKietzmann** [comment]
@OlegHahm, sure, but as it is hard to review or test these changes I wanted to see the compile tests being successful to avoid any kind of naming conflict or so... Please squash

**Comment by OlegHahm** [comment]
@PeterKietzmann, IMO that is not really mandatory for the release. You're free to reassign (maybe to @x3ro) and postone the milestone.

**Comment by OlegHahm** [comment]
On the other hand - if the CI agrees and the default example for MSB-A2 still builds, we should be fine.

**Comment by OlegHahm** [comment]
squashed

**Comment by PeterKietzmann** [comment]
Scrolled over it once more. Couldn't find something suspicious. Murdock agrees as well. So why not taking it in now? ACK and go

**Comment by OlegHahm** [comment]
Thanks!

**Pull request merged**